ath79: add support for Qxwlan E1700AC/E600G/E600GAC/E750A/E750G/E558

zhangsd20 opened the pull request:
E1700AC v2 based on Qualcomm/Atheros QCA9563 + QCA9880

Specification:

 - 750/400/250 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 3T3R 2.4 GHz
 - 3T3R 5 GHz
 - 2 x 10/1000M Mbps Ethernet (RJ45)
 - 1 x MiniPCI-e
 - 1 x SIM (3G/4G)
 - 1 x USB 2.0 Port
 - 5 x LED , 2 x Button(S8-Reset Buttun), 1 x power input
 - UART (J5) header on PCB (115200 8N1)

E600G v2 based on Qualcomm/Atheros QCA9531

Specification:

 - 650/600/200 MHz (CPU/DDR/AHB)
 - 128/64 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4 GHz
 - 2 x 10/100 Mbps Ethernet(RJ45)
 - 1 x MiniPCI-e
 - 1 x SIM (3G/4G)
 - 5 x LED , 1 x Button(SW2-Reset Buttun), 1 x power input
 - UART(J100) header on PCB(115200 8N1)

E600GAC v2 based on Qualcomm/Atheros QCA9531 + QCA9887

Specification:

  - 650/600/200 MHz (CPU/DDR/AHB)
  - 128/64 MB of RAM (DDR2)
  - 8/16 MB of FLASH (SPI NOR)
  - 2T2R 2.4 GHz
  - 1T1R 5 GHz
  - 2 x 10/100 Mbps Ethernet(RJ45)
  - 6 x LED (a three-color led), 2 x Button(SW2-Reset Buttun),1 x power input
  - UART (J100)header on PCB(115200 8N1)

Qxwlan E750A v4 is based on Qualcomm QCA9344.

Specification:

 - 560/450/225 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 5G GHz (AR9344)
 - 2x 10/100 Mbps Ethernet (one port with PoE support)
 - 1x miniPCIe slot (USB 2.0 bus only)
 - 7x LED (6 driven by GPIO)
 - 1x button (reset)
 - 1x DC jack for main power input (9-48 V)
 - UART (J23) and LEDs (J2) headers on PCB

Qxwlan E750G v8 is based on Qualcomm QCA9344 + QCA8334.

Specification:

 - 560/450/225 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4G GHz (AR9344)
 - 2x 10/100 Mbps Ethernet (PoE support)
 - 2x 10/100/1000 Mbps Ethernet
 - 7x LED (6 driven by GPIO)
 - 1x button (reset)
 - 1x DC jack for main power input (9-48 V)
 - UART (J23) and LEDs (J2) headers on PCB

Qxwlan E558 v2 is based on Qualcomm QCA9558 + AR8327.

Specification:

 - 720/600/200 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4 GHz (QCA9558)
 - 3x 10/100/1000 Mbps Ethernet (one port with PoE support)
 - 4x miniPCIe slot (USB 2.0 bus only)
 - 1x microSIM slot
 - 5x LED (4 driven by GPIO)
 - 1x button (reset)
 - 1x 3-pos switch
 - 1x DC jack for main power input (9-48 V)
 - UART (JP5) and LEDs (J8) headers on PCB

Flash instruction:

   1.Using tftp mode with UART connection and original LEDE image
      - Configure PC with static IP 192.168.1.10 and tftp server.
      - Rename "openwrt-ar71xx-generic-xxx-squashfs-sysupgrade.bin"
        to "firmware.bin" and place it in tftp server directory.
      - Connect PC with one of LAN ports, power up the router and press
        key "Enter" to access U-Boot CLI.
      - Use the following commands to update the device to LEDE:
        run lfw
      - After that the device will reboot and boot to LEDE.
      - Wait until all LEDs stops flashing and use the router.

   2.Using httpd mode with Web UI connection and original LEDE image
      - Configure PC with static IP 192.168.1.xxx(2-255) and tftp server.
      - Connect PC with one of LAN ports,press the reset button, power up
        the router and keep button pressed for around 6-7 seconds, until
        leds flashing.
      - Open your browser and enter 192.168.1.1,You will see the upgrade
        interface, select "openwrt-ar71xx-generic-xxx-squashfs-
        sysupgrade.bin" and click the upgrade button.
      - After that the device will reboot and boot to LEDE.
      - Wait until all LEDs stops flashing and use the router.

Signed-off-by:张鹏 <sd20@qxwlan.com>

zhangsd20 committed:
ath79: add support for Qxwlan E1700AC/E600G/E600GAC/E750A/E750G/E558

E1700AC based on Qualcomm/Atheros QCA9563 + QCA9880

Specification:

 - 750/400/250 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 3T3R 2.4 GHz
 - 3T3R 5 GHz
 - 2 x 10/100/1000 Mbps Ethernet (one port with PoE support)
 - 1 x MiniPCI-e
 - 1 x SIM (3G/4G)
 - 1 x USB 2.0 Port
 - 5 x LED , 2 x Button(S8-Reset Buttun), 1 x power input
 - UART (J5) header on PCB (115200 8N1)

E600G based on Qualcomm/Atheros QCA9531

Specification:

 - 650/600/200 MHz (CPU/DDR/AHB)
 - 128/64 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4 GHz
 - 2 x 10/100M Mbps Ethernetï(one port with PoE support)
 - 1 x MiniPCI-e
 - 1 x SIM (3G/4G)
 - 8 x LED , 1 x Button(SW2-Reset Buttun), 1 x power input
 - UART(J100) header on PCB(115200 8N1)

E600GAC based on Qualcomm/Atheros QCA9531 + QCA9887

Specification:

  - 650/600/200 MHz (CPU/DDR/AHB)
  - 128/64 MB of RAM (DDR2)
  - 8/16 MB of FLASH (SPI NOR)
  - 2T2R 2.4 GHz
  - 1T1R 5 GHz
  - 2 x 10/100M Mbps Ethernetï(one port with PoE support)
  - 6 x LED, 2 x Button(SW2-Reset Buttun),1 x power input
  - UART (J100)header on PCB(115200 8N1)

Qxwlan E750A v4 is based on Qualcomm QCA9344.

Specification:

 - 560/450/225 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 5G GHz (AR9344)
 - 2x 10/100 Mbps Ethernet (one port with PoE support)
 - 1x miniPCIe slot (USB 2.0 bus only)
 - 7x LED (6 driven by GPIO)
 - 1x button (reset)
 - 1x DC jack for main power input (9-48 V)
 - UART (J23) and LEDs (J2) headers on PCB

Qxwlan E750G v8 is based on Qualcomm QCA9344 + QCA9334.

Specification:

 - 560/450/225 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4G GHz (AR9344)
 - 2x 10/100/1000 Mbps Ethernet (one port with PoE support)
 - 7x LED (6 driven by GPIO)
 - 1x button (reset)
 - 1x DC jack for main power input (9-48 V)
 - UART (J23) and LEDs (J2) headers on PCB

Qxwlan E558 v2 is based on Qualcomm QCA9558 + AR8327.

Specification:

 - 720/600/200 MHz (CPU/DDR/AHB)
 - 128 MB of RAM (DDR2)
 - 8/16 MB of FLASH (SPI NOR)
 - 2T2R 2.4 GHz (QCA9558)
 - 3x 10/100/1000 Mbps Ethernet (one port with PoE support)
 - 4x miniPCIe slot (USB 2.0 bus only)
 - 2x microSIM slot
 - 5x LED (4 driven by GPIO)
 - 1x button (reset)
 - 1x DC jack for main power input (9-48 V)
 - UART (JP5) and LEDs (J8) headers on PCB

Flash instruction:

   1.Using tftp mode with UART connection and original LEDE image
      - Configure PC with static IP 192.168.1.10 and tftp server.
      - Rename "openwrt-ar71xx-generic-xxx-squashfs-sysupgrade.bin"
        to "firmware.bin" and place it in tftp server directory.
      - Connect PC with one of LAN ports, power up the router and press
        key "Enter" to access U-Boot CLI.
      - Use the following commands to update the device to LEDE:
        run lfw
      - After that the device will reboot and boot to LEDE.
      - Wait until all LEDs stops flashing and use the router.

   2.Using httpd mode with Web UI connection and original LEDE image
      - Configure PC with static IP 192.168.1.xxx(2-255) and tftp server.
      - Connect PC with one of LAN ports,press the reset button, power up
        the router and keep button pressed for around 6-7 seconds, until
        leds flashing.
      - Open your browser and enter 192.168.1.1,You will see the upgrade
        interface, select "openwrt-ar71xx-generic-xxx-squashfs-
        sysupgrade.bin" and click the upgrade button.
      - After that the device will reboot and boot to LEDE.
      - Wait until all LEDs stops flashing and use the router.

Signed-off-by: 张鹏 <sd20@qxwlan.com>

target/linux/ath79/generic/base-files/etc/board.d/01_leds

target/linux/ath79/generic/base-files/etc/board.d/01_leds

adschm commented:
Please split this down into multiple commits.
So, eventually, you probably end up with one matching "Specifications:" block per commit.
The DTSI files would then just be added with the first device needing them.

adschm commented on target/linux/ath79/dts/ar9344_qxwlan_e750a-v4-16m.dts:
```diff
@@ -0,0 +1,92 @@
+#include "ar9344_qxwlan_e750x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
```
These two includes can be dropped, they are already in the DTSI file (same for similar cases in other DTS files).

adschm commented on target/linux/ath79/dts/ar9344_qxwlan_e750a-v4-16m.dts:
```diff
@@ -0,0 +1,92 @@
+
+	mtd-mac-address = <&pridata 0x400>;
+	mtd-mac-address-increment = <1>;
+};
```
I'd prefer to have eth0 node before eth1 here (just stupidly move them).

This actually helps to prevent mistakes when people copy-paste stuff later.

Please do the same for the other files if relevant.

adschm commented on target/linux/ath79/dts/ar9344_qxwlan_e750a-v4-8m.dts:
```diff
@@ -0,0 +1,92 @@
+			partition@70000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x070000 0x790000>;
```
Personally, I'd create another DTSI - ar9344_qxwlan_e750a-v4.dtsi - which would contain everything except the last partition and model/compatible. The DTS would then just be

```
// SPDX-License-Identifier: GPL-2.0-or-later OR MIT

#include "ar9344_qxwlan_e750a-v4.dtsi"

/ {
	model = "Qxwlan E750A V4 8M";
	compatible = "qxwlan,e750a-v4-8m", "qca,ar9344";
};

&partitions {
	partition@70000 {
		compatible = "denx,uimage";
		label = "firmware";
		reg = <0x070000 0x790000>;
	};
};
```

adschm replied:
(Note that this means _keeping_ target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi as well, the new DTSI would just contain the contents of e750a-v4-xm dtses)

adschm commented on target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata:
```diff
@@ -22,6 +22,10 @@ case "$FIRMWARE" in
+	qxwlan,e1700ac-v2-8m|\
+	qxwlan,e1700ac-v2-16m|\
+	qxwlan,e600gac-v2-8m|\
+	qxwlan,e600gac-v2-16m|\
```
MAC address in available from caldata here, so the pri-data address in only required for ethernet?

zhangsd20 replied:
yes，you are right.

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1475,6 +1475,110 @@ define Device/qihoo_c301
+  SOC := qca9558
+  DEVICE_VENDOR := Qxwlan
+  DEVICE_MODEL := E558 V2 (16M flash)
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 -swconfig
```
kmod-usb-core can be dropped, it will be selected by kmod-usb2.

adschm replied:
Same applies below.

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1475,6 +1475,110 @@ define Device/qihoo_c301
+define Device/qxwlan_e1700ac-v2-16m
+  SOC := qca9563
+  DEVICE_VENDOR := Qxwlan
+  DEVICE_MODEL := E1700AC V2 (16M flash)
```
The flash size can be put into `DEVICE_VARIANT`, e.g.
```
  DEVICE_VENDOR := Qxwlan
  DEVICE_MODEL := E1700AC V2
  DEVICE_VARIANT := 16M
```

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1475,6 +1475,110 @@ define Device/qihoo_c301
+TARGET_DEVICES += qxwlan_e1700ac-v2-16m
+
+define Device/qxwlan_e1700ac-v2-8m
+  $(Device/qxwlan_e1700ac-v2-16m)
```
Please do not derive on device from another one. Create a shared node instead, and base both devices on it, e.g. Device/dlink_dap-13xx in the same file.

adschm commented on target/linux/ath79/generic/base-files/etc/board.d/02_network:
```diff
@@ -283,6 +287,12 @@ ath79_setup_interfaces()
+	qxwlan,e558-v2-16m)
+		ucidef_set_interfaces_lan_wan "eth1" "eth0"
+		ucidef_add_switch "switch0" \
+			"0u@eth1" "4:lan" "5:lan" "6u@eth0"  "3:wan"
```
Why untagged? That's quite uncommon.

adschm commented:
BTW: Thanks for already incorporating the recent changes from master like LED labels etc. and similar formal stuff.

adschm commented on target/linux/ath79/generic/base-files/etc/board.d/01_leds:
```diff
@@ -229,6 +229,15 @@ pcs,cr3000)
+qxwlan,e750a-v4-8m|\
+qxwlan,e750a-v4-16m)
+	ucidef_set_led_netdev "lan" "LAN" "green:lan" "eth0"
+	ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth1"
```
Typically, at least one of these needs to be ucidef_set_led_switch in order to properly detect link state changes.

adschm commented on target/linux/ath79/generic/base-files/etc/board.d/02_network:
```diff
@@ -283,6 +287,12 @@ ath79_setup_interfaces()
 		;;
+	qxwlan,e558-v2-8m|\
+	qxwlan,e558-v2-16m)
+		ucidef_set_interfaces_lan_wan "eth1" "eth0"
```
If the proper tagged variant of ucidef_add_switch is used, this line here can be dropped entirely.

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1475,6 +1475,110 @@ define Device/qihoo_c301
+  DEVICE_PACKAGES := kmod-usb2 kmod-ath10k-ct ath10k-firmware-qca988x-ct -swconfig
+  IMAGE_SIZE := 15936k
+  IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | \
+	append-metadata | check-size
```
This line is the same as in Device/Default, and thus can be dropped.

zhangsd20 commented:
Hi，
    How to  split this down into multiple commits？
    What should i do？



Thank you very much ！









在2020年10月14 22时14分，"Adrian Schmutzler"<notifications@github.com>写道：


Please split this down into multiple commits.
So, eventually, you probably end up with one matching "Specifications:" block per commit.
The DTSI files would then just be added with the first device needing them.
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
>       - Open your browser and enter 192.168.1.1,You will see the upgrade
>         interface, select â"openwrt-ar71xx-generic-xxx-squashfs-
>         sysupgrade.bin" and click the upgrade button.

There is an unmotivated "â" character in the flashing instructions (option 2), please remove it if you touch this anyway.

adschm commented:
I've picked the e750a-v4 and will apply it to master shortly for illustration (with tiny adjustments).

Please also take care to have commit author and signed-off-by consistent (can be fixed with `git commit --am --reset-author` or `git commit --am --author="Name <mail address>"`)

adschm commented:
https://github.com/openwrt/openwrt/commit/5666ca913a879e499dabb7364fff5e080abe8819

zhangsd20 commented:
ok，When I submitted, the Chinese I entered here was garbled, so I changed it to English.
I have changed it to "张鹏" in Chinese.


Thank you very much!















在2020年10月16 00时40分，"Adrian Schmutzler"<notifications@github.com>写道：


I've picked the e750a-v4 and will apply it to master shortly for illustration (with tiny adjustments).
Please also take care to have commit author and signed-off-by consistent (can be fixed with git commit --am --reset-author or git commit --am --author="Name <mail address>")
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Ok, I read your merged e750a, and I made the same modification to other models.
Thank you very much!













在2020年10月16 00时55分，"Adrian Schmutzler"<notifications@github.com>写道：


5666ca9
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
@zhangsd20 The specs for several devices (e.g. E600G(AC)) show flash 16/32 MB, but the devices all have 8/16m setup. Which is correct? Just report back here, so I can do the relevant changes myself then (unless it's non-trivial).

adschm commented:
Does the e600gac really have/need usb (can't tell from the specs) ...?

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
```diff
@@ -0,0 +1,48 @@
+
+&leds {
+	wlan {
+		label = "orange:wlan";
```
Which one is it and should we probably use amber:wlan2g or amber:wlan5g instead? (kernel prefers amber over orange ...)

zhangsd20 commented:
yes，you are right.
i will change it.











在2020年10月16 17时43分，"Adrian Schmutzler"<notifications@github.com>写道：


Does the e600gac really have/need usb (can't tell from the specs) ...?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
yes,it is wlan2g. i will change it













在2020年10月16 17时46分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
> +	model = "Qxwlan E600GAC v2 16M";
+	compatible = "qxwlan,e600gac-v2-16m", "qca,qca9531";
+};
+
+&keys {
+	wps {
+		label = "wps";
+		linux,code = <KEY_WPS_BUTTON>;
+		gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		debounce-interval = <60>;
+	};
+};
+
+&leds {
+	wlan {
+		label = "orange:wlan";Which one is it and should we probably use amber:wlan2g or amber:wlan5g instead? (kernel prefers amber over orange ...)
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Yes, there used to be 16/32, now it is changed to 8/16M, I forgot to modify the information


I just corrected the problem you pointed out













在2020年10月16 17时38分，"Adrian Schmutzler"<notifications@github.com>写道：


@zhangsd20 The specs for several devices (e.g. E600G(AC)) show flash 16/32 MB, but the devices all have 8/16m setup. Which is correct? Just report back here, so I can do the relevant changes myself then (unless it's non-trivial).
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
Thanks. Not terribly important, but the number of LEDs doesn't match either between commit message and DTS (I count 4 GPIO-controllable LEDs for e600g and 7 for e600gac?). Second button for e600gac is obviously WPS?

zhangsd20 commented:
yes，e600gac  have a three-color led light (3-gpio) and 1 wps button  .













在2020年10月16 18时05分，"Adrian Schmutzler"<notifications@github.com>写道：


Thanks. Not terribly important, but the number of LEDs doesn't match either between commit message and DTS (I count 4 GPIO-controllable LEDs for e600g and 7 for e600gac?). Second button for e600gac is obviously WPS?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented on target/linux/ath79/image/generic.mk:
```diff
@@ -1475,6 +1475,129 @@ define Device/qihoo_c301
+  SOC := ar9344
+  DEVICE_VENDOR := Qxwlan
+  DEVICE_MODEL := E750G
+  DEVICE_PACKAGES := kmod-usb2 -swconfig
```
I don't see any usb for this device, and swconfig is definitely required.

adschm commented on target/linux/ath79/generic/base-files/etc/board.d/02_network:
```diff
@@ -243,6 +243,10 @@ ath79_setup_interfaces()
+	qxwlan,e1700ac-v2-8m|\
+	qxwlan,e1700ac-v2-16m|\
+	qxwlan,e750g-v8-8m|\
+	qxwlan,e750g-v8-16m|\
```
>  - 2x 10/100 Mbps Ethernet (PoE support)
>  - 2x 10/100/1000 Mbps Ethernet

Which is true?

adschm commented on target/linux/ath79/dts/qca9558_qxwlan_e558.dtsi:
```diff
@@ -0,0 +1,165 @@
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
```
What about the 3-pos switch?

adschm commented on target/linux/ath79/dts/qca9558_qxwlan_e558.dtsi:
```diff
@@ -0,0 +1,165 @@
+			linux,default-trigger = "phy0tpt";
+		};
+
+	};
```
What about the fourth LED?

adschm commented on target/linux/ath79/dts/qca9558_qxwlan_e558.dtsi:
```diff
@@ -0,0 +1,165 @@
+		qss {
+			label = "green:qss";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
```
QSS LED should not be used for wifi status. Maybe this mix is the result of the "missing" LED?

adschm commented:
> 5 x LED , 2 x Button(S8-Reset Buttun), 1 x power input

I only see 1 button and 3 LEDs for e1700ac ...

zhangsd20 commented:
Hi,

    e750a/g  has a usb port, 4 pin headers lead out (J50)，e750g requires swconfig












在2020年10月16 18时15分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/image/generic.mk:
> +  IMAGE_SIZE := 15936k
+endef
+TARGET_DEVICES += qxwlan_e750a-v4-16m
+
+define Device/qxwlan_e750a-v4-8m
+  $(Device/qxwlan_e750a-v4)
+  DEVICE_VARIANT := v4 (8M)
+  IMAGE_SIZE := 7744k
+endef
+TARGET_DEVICES += qxwlan_e750a-v4-8m
+
+define Device/qxwlan_e750g-v8
+  SOC := ar9344
+  DEVICE_VENDOR := Qxwlan
+  DEVICE_MODEL := E750G
+  DEVICE_PACKAGES := kmod-usb2 -swconfigI don't see any usb for this device, and swconfig is definitely required.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
5 x LED：
     3 leds are controlled by gpio, one ath10k（5g  no need to define it, just select ath10k led）, one power supply


2 x button：
     One is gpio control software reset, the other is hardware reset










在2020年10月16 18时48分，"Adrian Schmutzler"<notifications@github.com>写道：


5 x LED , 2 x Button(S8-Reset Buttun), 1 x power input
I only see 1 button and 3 LEDs for e1700ac ...
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
sorry,  it is my fault, wlan corresponds to wifi status.
i will change it.














在2020年10月16 18时45分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9558_qxwlan_e558.dtsi:
> +
+		wlan {
+			label = "green:wlan";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "green:system";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		qss {
+			label = "green:qss";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";QSS LED should not be used for wifi status. Maybe this mix is the result of the "missing" LED?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Sorry, Specification is to copy the information from the previous submission.

I did not look carefully. I am very sorry that this has caused you trouble.













在2020年10月16 18时16分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/generic/base-files/etc/board.d/02_network:
> @@ -243,6 +243,10 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "2:lan:1" "3:lan:2" "4:lan:3" "5:lan:4" "1:wan"
 		;;
+	qxwlan,e1700ac-v2-8m|\
+	qxwlan,e1700ac-v2-16m|\
+	qxwlan,e750g-v8-8m|\
+	qxwlan,e750g-v8-16m|\2x 10/100 Mbps Ethernet (PoE support)2x 10/100/1000 Mbps EthernetWhich is true?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
And E750G v8 has five LEDs then, missing the lan/wan ones compared to E750A?

adschm commented:
> And E750G v8 has five LEDs then, missing the lan/wan ones compared to E750A?

Ah, the missing ones are these here:

```
static const struct ar8327_led_info e750g_v8_leds_qca8334[] = {
	AR8327_LED_INFO(PHY1_0, HW, "e750g-v8:green:lan"),
	AR8327_LED_INFO(PHY2_0, HW, "e750g-v8:green:wan"),
};
```

Do these work with ath79?

Apart from that, I just became aware that these devices were supported in ar71xx already. Is there a reason why you don't allow sysupgrade by adding the appropriate SUPPORTED_DEVICES strings? If not, I'd prepare a patch for that.

adschm commented:
I've pushed the current state here: https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/qxwlan2

E600G(AC) has more differences compared to ar71xx, so it's probably best to discuss them separately afterwards.

zhangsd20 commented:
E750G lan/wan  in       target/linux/ath79/generic/base-files/etc/board.d/02_network



If there is a conflict with ar71xx, we can use uboot to upgrade the firmware to ath79, and then use sysupgrade to upgrade.


It would be better if you can be compatible










在2020年10月17 19时30分，"Adrian Schmutzler"<notifications@github.com>写道：


And E750G v8 has five LEDs then, missing the lan/wan ones compared to E750A?
Ah, the missing ones are these here:
static const struct ar8327_led_info e750g_v8_leds_qca8334[] = {
	AR8327_LED_INFO(PHY1_0, HW, "e750g-v8:green:lan"),
	AR8327_LED_INFO(PHY2_0, HW, "e750g-v8:green:wan"),
};Do these work with ath79?
Apart from that, I just became aware that these devices were supported in ar71xx already. Is there a reason why you don't allow sysupgrade by adding the appropriate SUPPORTED_DEVICES strings? If not, I'd prepare a patch for that.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
E750G network led in ar9344_qxwlan_e750g-v8-16m.dts（&mdio0 ），work is ok ！





E750G lan/wan  in      target/linux/ath79/generic/base-files/etc/board.d/02_network



If there is a conflict with ar71xx, we can use uboot to upgrade the firmware to ath79, and then use sysupgrade to upgrade.


It would be better if you can be compatible










在2020年10月17 19时30分，"Adrian Schmutzler"<notifications@github.com>写道：


And E750G v8 has five LEDs then, missing the lan/wan ones compared to E750A?
Ah, the missing ones are these here:
static const struct ar8327_led_info e750g_v8_leds_qca8334[] = {
	AR8327_LED_INFO(PHY1_0, HW, "e750g-v8:green:lan"),
	AR8327_LED_INFO(PHY2_0, HW, "e750g-v8:green:wan"),
};Do these work with ath79?
Apart from that, I just became aware that these devices were supported in ar71xx already. Is there a reason why you don't allow sysupgrade by adding the appropriate SUPPORTED_DEVICES strings? If not, I'd prepare a patch for that.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Sorry, I don't understand what you mean.
What do you mean by the big difference? But I tested it on the motherboard and it works well,
and I haven’t seen any abnormalities yet









在2020年10月17 20时05分，"Adrian Schmutzler"<notifications@github.com>写道：


I've pushed the current state here: https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/qxwlan2
E600G(AC) has more differences compared to ar71xx, so it's probably best to discuss them separately afterwards.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
Okay, I've merged the devices except E600G(AC) already.

The remaining open question for this thread is whether I should allow upgrade from ar71xx by adding SUPPORTED_DEVICES, or not. See the topmost commit in https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/qxwlan2

For the remaining E600G/GAC, I'd be happy if you could resubmit that one properly, as discussing it based on the big commit here proves to be a little complicated at this stage.

zhangsd20 commented:
I can resubmit e600g/ac, but please tell me what is wrong with it,

and I will revise it and submit it again. Because I did not find any

problems in the test, so I don’t know how I should modify it to be correct.



Thank you very much !











在2020年10月19 20时14分，"Adrian Schmutzler"<notifications@github.com>写道：


Okay, I've merged the devices except E600G(AC) already.
The remaining open question for this thread is whether I should allow upgrade from ar71xx by adding SUPPORTED_DEVICES, or not. See the topmost commit in https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/qxwlan2
For the remaining E600G/GAC, I'd be happy if you could resubmit that one properly, as discussing it based on the big commit here proves to be a little complicated at this stage.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
If it is the number of leds, I can modify it to the same on ar71xx. But I want to explain why I write it differently. The number of LEDs we see on E600G is 5, but in fact, we have led more LEDs on j7 as a signal indicator. There are 6 LEDs we can see on the E600GAC, one of which is a tri-color LED, so it is 8.


I have modified the submitted information












在2020年10月19 20时14分，"Adrian Schmutzler"<notifications@github.com>写道：


Okay, I've merged the devices except E600G(AC) already.
The remaining open question for this thread is whether I should allow upgrade from ar71xx by adding SUPPORTED_DEVICES, or not. See the topmost commit in https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/qxwlan2
For the remaining E600G/GAC, I'd be happy if you could resubmit that one properly, as discussing it based on the big commit here proves to be a little complicated at this stage.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600g-v2-8m.dts:
```diff
@@ -0,0 +1,33 @@
+
+&leds {
+	wlan {
+		label = "blue:wlan";
```
ar71xx has this as blue:wan
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L47

So, ar71xx is wrong?

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600g-v2-16m.dts:
```diff
@@ -0,0 +1,34 @@
+&usb0 {
+	status = "okay";
+};
+
```
Trailing empty line should be removed. Same for 8m ...

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600g.dtsi:
```diff
@@ -0,0 +1,114 @@
+		compatible = "gpio-leds";
+
+		led_system: system {
+			label = "blue:system";
```
In ar71xx, this is called green:system for the the AC variant:
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L75

If that's correct, we should move the node (and the relevant aliases) to the DTS files.

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
```diff
@@ -0,0 +1,48 @@
+
+&leds {
+	wlan2g {
+		label = "orange:wlan2g";
```
Again, ar71xx labels this orange:wan
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L79

adschm commented on target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
```diff
@@ -0,0 +1,48 @@
+		linux,default-trigger = "phy1tpt";
+	};
+
+	control1 {
```
How about calling the nodes control_green, control_red and control_blue instead? Or does the number transport a relevant information?

adschm commented:
Okay, I've marked the differences with ar71xx above.
Apart from that, please add the compatibility measures with ar71xx to the final commit as just merged for the other devices:
https://github.com/openwrt/openwrt/commit/8087499ac50bc2358d5f9d872293a9a01fd64b5d

You should then be able to rebase and wrap up all remaining changes in a new PR for E600G(AC), so we can conclude this.

zhangsd20 commented:
yes，you can see on openwrt that there are only 2 Ethernet ports, one lan port and one wan port, so there is a green lan and a green wan.













在2020年10月27 03时54分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9531_qxwlan_e600g-v2-8m.dts:
> @@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca9531_qxwlan_e600g.dtsi"
+
+/ {
+	model = "Qxwlan E600G v2 8M";
+	compatible = "qxwlan,e600g-v2-8m", "qca,qca9531";
+};
+
+&leds {
+	wlan {
+		label = "blue:wlan";ar71xx has this as blue:wan
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L47
So, ar71xx is wrong?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
ar71xx is wrong, it should be wlan indicator






在2020年10月27 04时01分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
> +	model = "Qxwlan E600GAC v2 16M";
+	compatible = "qxwlan,e600gac-v2-16m", "qca,qca9531";
+};
+
+&keys {
+	wps {
+		label = "wps";
+		linux,code = <KEY_WPS_BUTTON>;
+		gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		debounce-interval = <60>;
+	};
+};
+
+&leds {
+	wlan2g {
+		label = "orange:wlan2g";Again, ar71xx labels this orange:wan
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L79
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
These lights can be defined by users, and currently we use them as signal strength indicators












在2020年10月27 04时02分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9531_qxwlan_e600gac-v2-16m.dts:
> +	wps {
+		label = "wps";
+		linux,code = <KEY_WPS_BUTTON>;
+		gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		debounce-interval = <60>;
+	};
+};
+
+&leds {
+	wlan2g {
+		label = "orange:wlan2g";
+		gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		linux,default-trigger = "phy1tpt";
+	};
+
+	control1 {How about calling the nodes control_green, control_red and control_blue instead? Or does the number transport a relevant information?
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Sorry, the definition of ar71xx is wrong. The system uses blue leds.



在2020年10月27 03时59分，"Adrian Schmutzler"<notifications@github.com>写道：


@adschm commented on this pull request.
In target/linux/ath79/dts/qca9531_qxwlan_e600g.dtsi:
> +	keys: keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+
+		led_system: system {
+			label = "blue:system";In ar71xx, this is called green:system for the the AC variant:
https://github.com/openwrt/openwrt/blob/openwrt-19.07/target/linux/ar71xx/files/arch/mips/ath79/mach-e600g-v2.c#L75
If that's correct, we should move the node (and the relevant aliases) to the DTS files.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
ok, thank you very much.
I have answered and modified the question you said












在2020年10月27 04时11分，"Adrian Schmutzler"<notifications@github.com>写道：


Okay, I've marked the differences with ar71xx above.
Apart from that, please add the compatibility measures with ar71xx to the final commit as just merged for the other devices:
8087499
You should then be able to rebase and wrap up all remaining changes in a new PR for E600G(AC), so we can conclude this.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

adschm commented:
In case that's not clear: I'm waiting for you to submit a _new PR_ with the updated changes for E600G(AC).

zhangsd20 commented:
哦库，i will creat new pr for e600g/ac













在2020年10月31 05时10分，"Adrian Schmutzler"<notifications@github.com>写道：


In case that's not clear: I'm waiting for you to submit a new PR with the updated changes for E600G(AC).
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

zhangsd20 commented:
Hi,
    https://github.com/openwrt/openwrt/pull/3550













在2020年10月31 05时10分，"Adrian Schmutzler"<notifications@github.com>写道：


In case that's not clear: I'm waiting for you to submit a new PR with the updated changes for E600G(AC).
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

Neustradamus commented:
Merged commits:

E750A v4:
- https://github.com/openwrt/openwrt/commit/5666ca913a879e499dabb7364fff5e080abe8819

E750G v8
- https://github.com/openwrt/openwrt/commit/27a7d071c0e0576eddeb641d1f68abe1b6205aee

E1700AC v2
- https://github.com/openwrt/openwrt/commit/4ff7bdfeebd4c974b74df74f779fc10891898f5d

E600G v2 / E600GAC v2
- https://github.com/openwrt/openwrt/commit/448de2e2e5c2c00a11bdf08147ee85242b2d48d2

Linked to:
- https://github.com/openwrt/openwrt/pull/3550
